src/model_state.jl

```diff
@@ -0,0 +1,150 @@
+"""
+    $TYPEDEF
+
+Represents the state of a "simulation" for a given `model`. `ModelState` consists of a
+`clock`, a `model`, and an initialized `StateVariables` data structure, as well as a `cache`
+for the timestepper and any relevant `inputs` provided by a corresponding `InputProvider`.
+The `ModelState` implements the `Oceananigans.AbstractModel` interface and can thus be
+treated as a "model" in `Oceananigans` `Simulation`s and output reading/writing utilities.
+"""
+struct ModelState{
```

Comment:
I already wrote it and told to you, I am fine with this. I think this is a good and pragmatic compromise, and I don't see an immediate downside.

Reply:
Just that I understand correctly. Currently we don't have a `Simulation` anymore, but we would probably also reuse or subtype Oceananigans' simulation to get the output to work, correct?

Reply:
Yes, we can re-use Oceananigans `Simulation` now. Running it works fine, but output isn't working yet because of some issues with the grid.